Scripts/Runtime/Entities/Tasks/Data/JobTaskWorkConfig.cs

```diff
@@ -40,12 +41,20 @@ internal JobTaskWorkConfig(ScheduleJobDelegate scheduleJobDelegate, AbstractTask
         /// <typeparam name="TKey">The type of the key</typeparam>
         /// <typeparam name="TInstance">The type of the data</typeparam>
         /// <returns>This <see cref="JobTaskWorkConfig"/> for chaining additional configuration.</returns>
-        public JobTaskWorkConfig ScheduleOn<TKey, TInstance>(VirtualData<TKey, TInstance> data, BatchStrategy batchStrategy)
-            where TKey : unmanaged, IEquatable<TKey>
-            where TInstance : unmanaged, IKeyedData<TKey>
+        public JobTaskWorkConfig ScheduleOn<TInstance>(VirtualData<TInstance> data, BatchStrategy batchStrategy)
+            where TInstance : unmanaged, IKeyedData
+        {
+            Debug_EnsureNoDuplicateScheduleInfo();
+            m_ScheduleInfo = new VirtualDataScheduleInfo<TInstance>(data, batchStrategy, false);
+            return this;
+        }
+        
+        //TODO: DISCUSS - CancelTaskWorkConfig vs JobTaskWorkConfig and abstract
```

Comment:
Should we separate into a CancelTaskWorkConfig? The more I'm working with this stuff on the developer end, the more it feels like we should lock down the types to only the functionality they should be used for.

`CancelTaskWorkConfig` -> `AbstractJobTaskWorkConfig` -> `AbstractTaskWorkConfig`
`JobTaskWorkConfig` -> `AbstractJobTaskWorkConfig` -> `AbstractTaskWorkConfig`
`MainTaskWorkConfig` -> `AbstractTaskWorkConfig`

Reply:
What's unique about cancel. Why can't it be a JobTaskWorkConfig?

Reply:
It can be, but then you are polluted with a bunch of functions that won't do anything/work

Reply:
ah ok, that seems fine I think